### PR TITLE
import: disable labels when necessary and update presets when initialized

### LINKED ISSUE #7854: import: "apply metadata" issues
A couple of possibly related issues with the new options in the import module:

If you tick "apply metadata", the metadata fields are active and can be selected, though for some reason, clicking on the "preset" drop-down doesn't do anything:

Screenshot_2021-01-14_23-33-10

If you un-tick "apply metadata", the metadata fields are deactivated, as expected. However, only the "preset" and "tags" labels are greyed out (to indicate that the fields are inactive). This should be changed so that all fields are visually inactive.

Screenshot_2021-01-14_23-34-50

darktable 3.5.0+639~gc77f1ff43 on ArchLinux

### PR DESCRIPTION
**Pull request by phweyland**
fix #7854 ~~(partially)~~

### CODE REVIEW

**Comment by phweyland** [comment]
@elstoc, I've added a "signal changed" to refresh the combobox at the end of metadata init_presets(), but I cannot test if that works.
Could please check if that solves the issue. Thanks in advance.

For metadata presets changes, there is currently no good trigger. We may create a DT_SIGNAL_PRESETS_CHANGED but for the time being I prefer focus on the issue.

**Comment by elstoc** [comment]
Thanks @phweyland that solves both of the issues in #7854.

Possibly a separate issue but can I ask a couple of questions about the presets. If I set the creator and then choose, for example, the "CC BY-SA" preset, it doesn't override the creator field but just _adds_ the "rights" field - is this intended behaviour? Also, there's no way to empty all of the metadata fields without manually clearing them one at a time. Perhaps a new hard-coded "None" preset would be useful or we could use the general module reset button. Happy to raise a FR for these but wanted to ask the question first.

**Comment by TurboGit** [comment]
> Possibly a separate issue but can I ask a couple of questions about the presets. If I set the creator and then choose, for example, the "CC BY-SA" preset, it doesn't override the creator field but just _adds_ the "rights" field - is this intended behaviour?

No, it is not intended.

> Perhaps a new hard-coded "None" preset

No, the reset button on the import module should be used for that.

**Comment by TurboGit** [comment]
> Possibly a separate issue but can I ask a couple of questions about the presets. If I set the creator and then choose, for example, the "CC BY-SA" preset, it doesn't override the creator field but just _adds_ the "rights" field - is this intended behaviour?

In fact for me whatever CC-BY I select only the **rights** field is set even if all others empty.

**Comment by phweyland** [comment]
> that solves both of the issues in #7854.

 :+1: 





> but just _adds_ the "rights" field

I'll look at this. This should be easier.

**Comment by elstoc** [comment]
FYI the metadata module behaves the same way (i.e. doesn't overwrite already-populated fields)

**Comment by TurboGit** [comment]
> FYI the metadata module behaves the same way (i.e. doesn't overwrite already-populated fields)

For me even empty fields are not overwritten and only the rights fields is set. Do you have other fields set when empty? Maybe there is two different bugs? But bet @phweyland will fix that :)

**Comment by phweyland** [comment]
> > but just _adds_ the "rights" field
> 
> I'll look at this. This should be easier.

> In fact for me whatever CC-BY I select only the rights field is set even if all others empty.

From my standpoint this works well.

The hard-coded presets only define the rights. 
If you apply the presets, only the rights field is updated. As far as I know it was like this before.

Now, if you create a user metadata preset, the fields you define in the presets are applied as well.

I think this is correct anyway, like this you can set some metadata manually and complete with the presets without loosing your manual inputs.

**Comment by phweyland** [comment]
> For me even empty fields are not overwritten and only the rights fields is set

I haven't changed anything there. And I think this is correct.

**Comment by phweyland** [comment]
> "None" preset would be useful or we could use the general module reset button.

Suggestion for that: double click on "preset" label to clear all fields. Would that be intuitive enough ?

**Comment by phweyland** [comment]
> manually clearing them one at a time.

Are you aware that a double click on the label clears the entry ?

**Comment by elstoc** [comment]
> Are you aware that a double click on the label clears the entry ?

No. Perhaps needs a tooltip (and might also be useful to duplicate this functionality in the metadata module). Alternatively a clear icon (like in the module search) would be more easily discoverable, though I'm not sure how this would interact with scroll bars.

**Comment by phweyland** [comment]
> Perhaps needs a tooltip (and might also be useful to duplicate this functionality in the metadata module). 

I'll do it for import. 
I take note for metadata module.

**Comment by phweyland** [comment]
Clear all added + tooltip.
Tested with new db but not with a full one.
With the last rebase the database upgrade stucks.

**Comment by phweyland** [comment]
@elstoc. A big thanks, without your kind help I would not have fixed this one. :smiley:

**Comment by elstoc** [comment]
My pleasure. Difficult to fix when you can't reproduce things.

File: src/gui/import_metadata.c [hunk 1]
```diff
@@ -196,34 +209,63 @@ void dt_import_metadata_init(dt_import_metadata_t *metadata)
...
-  gtk_grid_attach(GTK_GRID(grid), label, 0, 0, 1, 1);
-  gtk_widget_set_tooltip_text(GTK_WIDGET(label), _("metadata to be applied per default"));
+  GtkWidget *labelev = gtk_event_box_new();
+  gtk_widget_set_visible(labelev, TRUE);
+  gtk_widget_add_events(labelev, GDK_BUTTON_PRESS_MASK);
+  gtk_container_add(GTK_CONTAINER(labelev), label);
+  gtk_grid_attach(GTK_GRID(grid), labelev, 0, 0, 1, 1);
+  gtk_widget_set_tooltip_text(GTK_WIDGET(label), _("metadata to be applied per default"
+                                                   "\ndouble-click on a label to clear the corresponding entry"
+                                                   "\ndouble-click on 'preset' to clear all entries"));
```
Comment (TurboGit) [inline src/gui/import_metadata.c:256]: I'm not ok for this one, as least not a the only solution. We have a reset button on the import dialog it seems better to use it to clear all entries (reset to default for all entries in the import dialog and empty all metadata).

File: src/gui/import_metadata.c [hunk 2]
```diff
@@ -196,34 +209,63 @@ void dt_import_metadata_init(dt_import_metadata_t *metadata)
...
   gtk_widget_set_halign(label, GTK_ALIGN_START);
-  gtk_grid_attach(GTK_GRID(grid), label, 0, 0, 1, 1);
-  gtk_widget_set_tooltip_text(GTK_WIDGET(label), _("metadata to be applied per default"));
+  GtkWidget *labelev = gtk_event_box_new();
+  gtk_widget_set_visible(labelev, TRUE);
+  gtk_widget_add_events(labelev, GDK_BUTTON_PRESS_MASK);
+  gtk_container_add(GTK_CONTAINER(labelev), label);
+  gtk_grid_attach(GTK_GRID(grid), labelev, 0, 0, 1, 1);
+  gtk_widget_set_tooltip_text(GTK_WIDGET(label), _("metadata to be applied per default"
+                                                   "\ndouble-click on a label to clear the corresponding entry"
```
Comment (TurboGit) [inline src/gui/import_metadata.c:255]: The same action on the metadata editor does nothing, I bet will have request to do the same for consistency.

**Review by TurboGit — CHANGES_REQUESTED** [review]
Some tidy-up needed.

File: src/gui/import_metadata.c (hunk 1 above)
Comment (elstoc) [reply src/gui/import_metadata.c:256]: I'm not sure. I suspect people are likely to want to clear metadata independently of the other settings (which are more likely to remain static). Though  double-clicking on labels to clear text isn't the most discoverable thing, and an icon to the right of the entry box is more standard.

Comment (TurboGit) [reply src/gui/import_metadata.c:256]: Well the (reset) icon on any module currently reset the parameters to default, I'm proposing just that and keeping whatever other clear options.

Comment (elstoc) [reply src/gui/import_metadata.c:256]: It's not 100% consistent across lib modules but yes if we're going to enable the reset button that's what it should do.

Comment (phweyland) [reply src/gui/import_metadata.c:256]: > Though double-clicking on labels to clear text isn't the most discoverable thing, and an icon to the right of the entry box is more standard.

Would be easy to do, but we lack of space. 

> Well the (reset) icon on any module currently reset the parameters to default, I'm proposing just that and keeping whatever other clear options.

Maybe something to take in account: the preferences "session options" could come here too at some point.
Module reset and metadata reset can make the difference here.

I add the reset button.

File: src/gui/import_metadata.c (hunk 2 above)
Comment (phweyland) [reply src/gui/import_metadata.c:255]: I was keeping that for another PR, but I can add this too.

File: src/gui/import_metadata.c (hunk 1 above)
Comment (TurboGit) [reply src/gui/import_metadata.c:256]: Thanks a lot Philippe !

File: src/gui/import_metadata.c (hunk 2 above)
Comment (phweyland) [reply src/gui/import_metadata.c:269]: > The same action on the metadata editor does nothing, I bet will have request to do the same for consistency.

Yes, it does nothing. But should it reset the image(s) metadata ?

Reset button should reset the preferences of the module (visible, private metadata). Only.

The button clear (already in place) clears metadata values.
I'll add the double-click on the label to clear the corresponding metadata.

OK ?

Comment (elstoc) [reply src/gui/import_metadata.c:269]: I've raised #7859 to discuss the reset button on lib modules - imo it's not really very consistent or obvious so perhaps the reset button action is better discussed there?

**Comment by phweyland** [comment]
In the last commit I've modified metadata:
- reset button reset the pref (show all by default metadata)
- double-click on label reset the metadata.

Of course, I can always unwind something ...

**Comment by phweyland** [comment]
@turbogit,
Pascal I suggest to remove the metadata module changes for which there are some discussions and let only the fixes of import in this PR.
I'll open another for the metadata module.
OK ?

**Comment by TurboGit** [comment]
Works for me Philippe.

**Comment by phweyland** [comment]
Split is done. New PR #7905.

**Comment by TurboGit** [comment]
Code looks good, I still need to test.

**Review by TurboGit — APPROVED** [review]
Thanks !

**Pull request merged**